core/src/test/java/org/everit/json/schema/internal/DefaultFormatValidatorTest.java

```diff
@@ -44,18 +44,64 @@ private void assertSuccess(final String subject, final FormatValidator format) {
     @Test
     public void dateTimeExceedingLimits() {
         assertFailure("1996-60-999T16:39:57-08:00", new DateTimeFormatValidator(),
-                "[1996-60-999T16:39:57-08:00] is not a valid date-time");
+                "[1996-60-999T16:39:57-08:00] is not a valid date-time. Expected [yyyy-MM-dd'T'HH:mm:ssZ, yyyy-MM-dd'T'HH:mm:ss.[0-9]{1,9}Z]");
     }
 
     @Test
     public void dateTimeFormatFailure() {
         assertFailure("2015-03-13T11:00:000", new DateTimeFormatValidator(),
-                "[2015-03-13T11:00:000] is not a valid date-time");
+                "[2015-03-13T11:00:000] is not a valid date-time. Expected [yyyy-MM-dd'T'HH:mm:ssZ, yyyy-MM-dd'T'HH:mm:ss.[0-9]{1,9}Z]");
+    }
+
+    @Test
+    public void dateTimeWithSingleDigitInSecFracSuccess() {
+        assertSuccess("2015-02-28T11:00:00.1Z", new DateTimeFormatValidator());
+    }
+
+    @Test
+    public void dateTimeWithTwoDigitsInSecFracSuccess() {
+        assertSuccess("2015-02-28T11:00:00.12Z", new DateTimeFormatValidator());
+    }
+
+    @Test
+    public void dateTimeWithThreeDigitsInSecFracSuccess() {
+        assertSuccess("2015-02-28T11:00:00.123Z", new DateTimeFormatValidator());
+    }
+
+    @Test
+    public void dateTimeWithFourDigitsInSecFracSuccess() {
+        assertSuccess("2015-02-28T11:00:00.1234Z", new DateTimeFormatValidator());
+    }
+
+    @Test
+    public void dateTimeWithFiveDigitsInSecFracSuccess() {
+        assertSuccess("2015-02-28T11:00:00.12345Z", new DateTimeFormatValidator());
+    }
+
+    @Test
+    public void dateTimeWithSixDigitsInSecFracSuccess() {
+        assertSuccess("2015-02-28T11:00:00.123456Z", new DateTimeFormatValidator());
+    }
+
+    @Test
+    public void dateTimeWithSevenDigitsInSecFracSuccess() {
+        assertSuccess("2015-02-28T11:00:00.1234567Z", new DateTimeFormatValidator());
+    }
+
+    @Test
+    public void dateTimeWithEightDigitsInSecFracSuccess() {
+        assertSuccess("2015-02-28T11:00:00.12345678Z", new DateTimeFormatValidator());
+    }
+
+    @Test
+    public void dateTimeWithNineDigitsInSecFracSuccess() {
+        assertSuccess("2015-02-28T11:00:00.123456789Z", new DateTimeFormatValidator());
     }
 
     @Test
-    public void dateTimeSecFracSuccess() {
-        assertSuccess("2015-02-28T11:00:00.111Z", new DateTimeFormatValidator());
```

Comment:
Ok, found it.
One final note: I've just found in the [RFC3339 document](https://www.ietf.org/rfc/rfc3339.txt) that their ABNF defines fractions as:

    time-fraction     = ("," / ".") 1*DIGIT

I had a quick glance at the javadoc and I couldn't find any easy way to support comma too, not only dot as a separator character. Do you think there is any simple solution to fix it?